### PR TITLE
Add uv_thread_id(). Modify uv_thread_t on Windows to have the thread id

### PR DESCRIPTION
**Pull request by hnakamur**
as well as the thread handle.

### CODE REVIEW

File: src/uv-common.c [hunk 1]
```diff
@@ -301,8 +301,9 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
   ctx->arg = arg;
 
 #ifdef _WIN32
-  *tid = (HANDLE) _beginthreadex(NULL, 0, uv__thread_start, ctx, 0, NULL);
-  if (*tid == 0) {
+  tid->thread = (HANDLE) _beginthreadex(NULL, 0, uv__thread_start, ctx, 0,
+    &tid->thrdaddr);
```
Comment (bnoordhuis) [inline src/uv-common.c:305]: Style: arguments need to line up.

**Comment by bnoordhuis** [comment]
I'm not against this change per se but can you explain why you want it?

**Comment by hnakamur** [comment]
I thought we can use uv_thread_id() to check if the thread is self, like:
uv_thread_id(tid) == uv_thread_self().

However, I found we need to use pthread_equal() to check if two threads are equal.
http://pubs.opengroup.org/onlinepubs/007904975/functions/pthread_equal.html

Now I think it may be better to add uv_thread_equal(uv_thread_t\* t1, uv_thread_t\* t2) instead of adding uv_thread_id().
But then I don't understand the supposed usage of uv_thread_self(). Could you explain why it is added?

**Comment by bnoordhuis** [comment]
> But then I don't understand the supposed usage of uv_thread_self(). Could you explain why it is added?

node.js uses it for its openssl bindings: https://github.com/joyent/node/blob/1f76a2e/src/node_crypto.cc#L88

**Comment by hnakamur** [comment]
Thanks for explanation. I cancel this pull request.
Sorry to bother you.

**Comment by bnoordhuis** [comment]
No problem. Yours is the more correct approach actually but uv_thread_self() is somewhat easier in everyday use.

**Pull request closed without merging**